content/articles/oop-in-java-abstraction-and-encapsulation/index.md

```diff
@@ -1,350 +1,482 @@
-        this.color = color;
-        this.breed = breed;
-    }
+   private String color;
 
-    public boolean isSleeping(){
-        System.out.println("Sleeping");
-        return true;
-    }
+   private String breed;
 
-    public boolean isEating(){
-        System.out.println("Eating");
-        return true;
-    }
+   //Dog arg constructor
 
-    public boolean isbarking(){
-        System.out.println("Dog barking...");
-        return true;
-    }
+   public Dog(String name, int age, String color, String breed) {
+
+       this.name = name;
+
+       this.age = age;
+
+       this.color = color;
+
+       this.breed = breed;
+
+   }
+
+   public boolean isSleeping(){
+
+       System.out.println("Sleeping");
+
+       return true;
+
+   }
+
+   public boolean isEating(){
+
+       System.out.println("Eating");
+
+       return true;
+
+   }
+
+   public boolean isbarking(){
+
+       System.out.println("Dog barking...");
+
+       return true;
+
+   }
+
+   public boolean isRunning(){
+
+       System.out.println("Dog running");
+
+       return true;    
+
+   }   
 
-    public boolean isRunning(){
-        System.out.println("Dog running");
-        return true;	
-    }   
 }
+
 ```
 
 To create a new dog object
 
 ```java
+
 public class Dog_Main {
-   public static void main(String[] args) {
-       Dog littleDog = new Dog("Bingo", 6,"Brown", "German Shepherd");
-       System.out.println("Is the dog running:? "+ littleDog.isrunning());
 
-       System.out.println();
+  public static void main(String[] args) {
 
-       System.out.println("is the dog sleeping:? " + littleDog.isSleeping());
+      Dog littleDog = new Dog("Bingo", 6,"Brown", "German Shepherd");
 
-       System.out.println();
+      System.out.println("Is the dog running:? "+ littleDog.isrunning());
 
-       System.out.println("Is the dog barking:? " + littleDog.isbarking());
+      System.out.println();
 
-       System.out.println();
+      System.out.println("is the dog sleeping:? " + littleDog.isSleeping());
 
-       System.out.println("Is the dog eating:? " + littleDog.iseating());
+      System.out.println();
 
-       System.out.println();
+      System.out.println("Is the dog barking:? " + littleDog.isbarking());
+
+      System.out.println();
+
+      System.out.println("Is the dog eating:? " + littleDog.iseating());
+
+      System.out.println();
+
+      System.out.println("The dog name is:  " + littleDog.name);
+
+  }
 
-       System.out.println("The dog name is:  " + littleDog.name);
-   }
 }
+
 ```
 
 Now you see, we’ve created a new dog with the name Bingo, you can go on to create as many dogs with different states or attributes.
 
-
 Here is the output
 
 Dog running
+
 Is the dog running:? true
 
 Sleeping
+
 is the dog sleeping:? true
 
 Dog barking...
+
 Is the dog barking:? true
 
 Eating
+
 Is the dog eating:? true
 
 The dog name is: Bingo
 
 Now that we’ve got this out of the way, let's talk about Abstraction and Encapsulation properly.
```

Comment:
Don't capitalize the words *abstraction* and *encapsulation* as talked about earlier.